pkgs/by-name/op/opencode/package.nix

```diff
@@ -1,98 +1,173 @@
 {
   lib,
+  stdenv,
   stdenvNoCC,
-  fetchurl,
-  autoPatchelfHook,
-  common-updater-scripts,
-  curl,
-  installShellFiles,
-  jq,
-  unzip,
+  bun,
+  cacert,
+  fetchFromGitHub,
+  fzf,
+  makeBinaryWrapper,
+  models-dev,
+  nix-update-script,
+  nodejs,
+  ripgrep,
   testers,
-  writeShellScript,
+  writableTmpDirAsHomeHook,
 }:
 stdenvNoCC.mkDerivation (finalAttrs: {
   pname = "opencode";
-  version = "1.0.10";
+  version = "1.0.15";
+  src = fetchFromGitHub {
+    owner = "sst";
+    repo = "opencode";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-2iZOw6oOrmzIKWdCe+nLjpWtRyS4YDmafjfALwWgYtE=";
+  };
+
+  src-with-node_modules = stdenvNoCC.mkDerivation {
```

Comment:
I think that moving this definition to a `let in` block above `stdenvNoCC.mkDerivation (finalAttr: {` would enhance readability.